### PR TITLE
STAC API initial skeleton

### PR DESCRIPTION
**Pull request by aaime**
Not much to get excited about, has landing page, conformance, API. Period. :-D
Actual functionality to follow later.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACService.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    @GetMapping(name = "getLandingPage")
+    @ResponseBody
+    @HTMLResponseBody(templateName = "landingPage.ftl", fileName = "landingPage.html")
+    public STACLandingPage getLandingPage() throws IOException {
+        return new STACLandingPage(getService(), "ogc/stac");
+    }
+
+    @GetMapping(path = "conformance", name = "getConformanceDeclaration")
+    @ResponseBody
+    @HTMLResponseBody(templateName = "conformance.ftl", fileName = "conformance.html")
```
Comment (bradh) [inline src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACService.java:83]: Note that this stuff has to get repeated in the landing page too. 

https://github.com/radiantearth/stac-api-spec/tree/master/core#stac-api---core-specification :
"Note the conformsTo JSON object follows exactly the structure of OGC API - Features declaration of conformance classes, except is available directly under the landing page. This is a slight break from how OGC API does things, as STAC feels it is important for clients to be able to understand conformance in a single request. To be conformant to OGC API's the /conformance endpoint must be implemented as well."

Comment (aaime) [reply src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACService.java:83]: Oddball... but in the spec indeed. Fixed.

File: src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACAPIBuilder.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        // the external documentation
+        api.externalDocs(
+                new ExternalDocumentation()
+                        .description("STAC API specification")
+                        .url("https://github.com/radiantearth/stac-api-spec"));
+
+        // adjust path output formats
+        declareGetResponseFormats(api, "/", OpenAPI.class);
+        declareGetResponseFormats(api, "/conformance", ConformanceDocument.class);
+        // TODO: these needs to be ajusted once we have
```
[the remark this answers is not in the document]
Comment (aaime) [reply src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACAPIBuilder.java]: Fixed

File: src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACLandingPage.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+package org.geoserver.ogcapi.stac;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import org.geoserver.ogcapi.AbstractLandingPageDocumentNoConformance;
+import org.geoserver.ogcapi.ConformanceDocument;
+import org.geoserver.ogcapi.Link;
+import org.geoserver.opensearch.eo.OSEOInfo;
+
+/** A Features server landing page */
```
[the remark this answers is not in the document]
Comment (aaime) [reply src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACLandingPage.java]: Fixed

File: src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACService.java (hunk 1 above)
Comment (bradh) [reply src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/stac/STACService.java:83]: Oddball enough I thought I should quote both reference and text! If I hadn't seen it, might not have believed it...

**Pull request merged**